src/dune_rules/gen_rules.ml

```diff
@@ -146,6 +146,82 @@ end = struct
       | true ->
         let+ () = Mdx.gen_rules ~sctx ~dir ~scope ~expander mdx in
         empty_none)
+    | Melange mel ->
```

Comment:
Could you create a module called `Melange_rules` and put all this code there?

Reply:
Done in https://github.com/ocaml/dune/pull/6349/commits/270b83e480db47432100fbae7035570bcfed51f4.